src/Components/Aspire.StackExchange.Redis/AspireRedisExtensions.cs

```diff
@@ -80,17 +80,14 @@ public static class AspireRedisExtensions
 
         configureSettings?.Invoke(settings);
 
+        var optionsName = serviceKey is null ? Options.Options.DefaultName : connectionName;
+
         // see comments on ConfigurationOptionsFactory for why a factory is used here
-        builder.Services.TryAdd(ServiceDescriptor.Transient(typeof(IOptionsFactory<ConfigurationOptions>),
-            sp => new ConfigurationOptionsFactory(
-                settings,
-                sp.GetServices<IConfigureOptions<ConfigurationOptions>>(),
-                sp.GetServices<IPostConfigureOptions<ConfigurationOptions>>(),
-                sp.GetServices<IValidateOptions<ConfigurationOptions>>())));
-
-        string? optionsName = serviceKey is null ? null : connectionName;
+        builder.Services.AddTransient(sp => new RedisOptionsSettingsPair { OptionsName = optionsName, Settings = settings });
```

Comment:
This could use keyed di itself

Reply:
How? The IOptions system doesn't work with keyed DI.

Reply:
Isn't that how it works?
```csharp
builder.Services.AddKeyedTransient<IOptions<Foo>>("key", (_, _) => Options.Create(new Fo0()));
builder.Services.AddKeyedTransient<IOptions<Foo>>("key2", (_, _) => Options.Create(new Fo0()));
```

Reply:
OK, I see what you mean. Just the `RedisOptionsSettingsPair` uses keyed DI, but not the OptionsFactory. I thought you meant the whole factory as well.